src/telemetry/storage_usage.rs

```diff
@@ -0,0 +1,79 @@
+/*
+ * This file is part of Edgehog.
+ *
+ * Copyright 2022 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use crate::error::DeviceManagerError;
+use astarte_sdk::types::AstarteType;
+use serde::Serialize;
+use std::collections::HashMap;
+use sysinfo::{DiskExt, System, SystemExt};
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct DiskUsage {
+    pub disk_name: String,
+    pub total_bytes: i64,
+    pub free_bytes: i64,
+}
+
+/// get available partition disks on the system
+fn get_disks() -> Result<Vec<DiskUsage>, DeviceManagerError> {
+    let mut ret: Vec<DiskUsage> = Vec::new();
+    let mut sys = System::new_all();
+    sys.refresh_disks();
+
+    for disk in sys.disks() {
+        let mut map: HashMap<String, AstarteType> = HashMap::new();
+        let disk_name = disk.name().to_str().unwrap().replace("/dev", "");
+        map.insert(
+            format!("{}/totalBytes", disk_name),
+            (disk.total_space() as i64).into(),
+        );
+        map.insert(
+            format!("{}/freeBytes", disk_name),
+            (disk.available_space() as i64).into(),
+        );
```

Comment:
Remove this part, you have already converted the Disk Usage to Astarte properties in `get_storage_usage` function, also the map definition at 42 line.

Reply:
removed